lib/parsers/gomod-version-directive-parser.ts

```diff
@@ -0,0 +1,28 @@
+import { MissingGoModVersionDirectiveError } from '../errors';
+
+// Parse the Go version directive from the go.mod file, if defined.
+// The logic is more or less identical to the actual `go` command's. See more here:
+// https://cs.opensource.google/go/x/mod/+/7c05a442b7c1d1a107879b4a090bb5a38d3774a1:modfile/rule.go;l=342-362
+const GoVersionRE = /^go\s(?<version>([1-9][0-9]*)\.(0|[1-9][0-9]*))/m;
+
+// Attempt to parse the go directive out of a go.mod file, either in strict mode or not.
+// If strict mode, will cause a failure if no version is found, otherwise just return undefined.
+// Both options are available as different versions of the go cli tool have acted differently around this issue.
+export function parseGoModVersionDirective(
+  goModFileContent: string,
+  strict = true,
+): string | undefined | MissingGoModVersionDirectiveError {
```

Comment:
Can the error be returned from this function? I don't see that represented in the tests.

Reply:
Isn't this it: https://github.com/snyk/snyk-go-parser/pull/40/files#diff-2aa9d31d6185455203535371b1a78e7b2fda69d0f055e9d0cf572b4c9ead5d2eR59-R63 ?

Reply:
Yes, but that is checking a throws not that it was returned. Declaring the error in the function signature indicates it can be returned. There is no way to indicate a function throws a particular kind of error.